### PR TITLE
Pick data table framework

### LINKED ISSUE #65: Select a data table framework
ryans research: https://github.com/ctoec/data-collection/issues/51

### PR DESCRIPTION
**Pull request by JordanGuinn**
Closes #65.  Also includes a very, **very** rough sketch of an example table.

### CODE REVIEW

File: client/src/containers/CheckData/CheckData.tsx [hunk 1]
```diff
@@ -1,7 +1,84 @@
...
+import FirstPage from '@material-ui/icons/FirstPage';
+import LastPage from '@material-ui/icons/LastPage';
+import Remove from '@material-ui/icons/Remove';
+import SaveAlt from '@material-ui/icons/SaveAlt';
+import Search from '@material-ui/icons/Search';
+import ViewColumn from '@material-ui/icons/ViewColumn';
+
+import { createMuiTheme, MuiThemeProvider } from '@material-ui/core';
+
+const tableIcons = {
```
Comment (JordanGuinn) [inline client/src/containers/CheckData/CheckData.tsx:23]: Now realizing I need to move this elsewhere.

**Comment by JordanGuinn** [comment]
Will be messing with some of the options a bit more within this PR, at which point I'll fully satisfy the eslint rules currently causing the front-end build to fail.

File: client/src/containers/CheckData/CheckData.tsx [hunk 2]
```diff
@@ -1,7 +1,84 @@
...
+      { title: 'Surname', field: 'surname' },
+      { title: 'Birth Year', field: 'birthYear', type: 'numeric' },
+      {
+        title: 'Birth Place',
+        field: 'birthCity',
+        lookup: { 34: 'Somewhere', 63: 'Somewhere Else' },
+      },
+    ]}
+    data={[
+      { name: 'Michael', surname: 'Jordan', birthYear: 1987, birthCity: 63 },
```
[the remark this answers is not in the document]
Comment (JordanGuinn) [reply client/src/containers/CheckData/CheckData.tsx:72]: Not gonna lie - it _definitely_ took longer than it should've to come up with this fake data 😅

[the remark this answers is not in the document]
Comment (jlhogan) [reply client/src/containers/CheckData/CheckData.tsx:72]: LeBron is the MOST FAMOUS 3 year old ever and we all know that..

**Comment by JordanGuinn** [comment]
Yes and yes!  Both individual edits and batch operations are available to us, as well as custom styling of header/data rows and/or columns as needed.  There does seem to be [an outstanding issue](https://github.com/mbrn/material-table/issues/667) with trying to fully write and deploy our own custom Material theme, but that shouldn't be too much of an issue assuming we're not looking to _fundamentally rewrite_ anything stylistically.

**Comment by cailyncodes** [comment]
Fyi that trying to edit doesn't seem to actually update the data

**Comment by JordanGuinn** [comment]
Right - it's not hooked up to do that in it's current state.  Once we have an underlying data source that isn't just a static array, we should be able to take advantage of the library's [update hooks](https://github.com/ctoec/data-collection/blob/42d5dfffe4a998843e85bd88625eba8abeb66ca1/client/src/containers/CheckData/CheckData.tsx#L74) and actually do something with those events.

**Comment by cailyncodes** [comment]
Gotcha!

**Pull request merged**